File: code/746.min-cost-climbing-stairs.php
<?php
/*
 * @lc app=leetcode id=746 lang=php
 *
 * [746] Min Cost Climbing Stairs
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $cost
     * @return Integer
     */
    function minCostClimbingStairs($cost) {
        $answers = [];
        $answers[count($cost)] = 0;
        $answers[count($cost) + 1] = 0;
        for ($i = count($cost) - 1; $i >= 0; $i--) {
            $answers[$i] = $cost[$i] + min($answers[$i + 1], $answers[$i + 2]);
        }

        return min($answers[0], $answers[1]);
    }
}
// @lc code=end

File: code/746.min-cost-climbing-stairs.go
package code

/*
 * @lc app=leetcode id=746 lang=golang
 *
 * [746] Min Cost Climbing Stairs
 */

// @lc code=start
func minCostClimbingStairs(cost []int) int {
	ans := make([]int, len(cost))
	ans[0] = cost[0]
	ans[1] = cost[1]
	for i := 2; i < len(cost); i++ {
		ans[i] = cost[i] + min(ans[i-1], ans[i-2])
	}

	return min(ans[len(cost)-1], ans[len(cost)-2])
}

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}

// @lc code=end
